### ISSUE TITLE
Specific certain "Topic Client" causes gives error when opening to view queries

### ISSUE THREAD

**Issue by Hydranet** [pi-hole/pi-hole#4209]
### Versions

- Pi-hole:  v5.3.1
- AdminLTE:  v5.5
- FTL:  v5.8.1

### Platform

- OS and version:  RHEL Clone 8.4

- Platform:  Podman

### Expected behavior

When I click on a top client(Total) I expect to be able to view all the queries.
### Actual behavior / bug

When I open the top client to view the queries it gives an error: "An Unknown error occured while loading data"
### Steps to reproduce

1. Go to Dashboard
2. Click on the Top Clients(Total) (in my case pihole)
3. Error message pops-up: "An unknown  error occured while loading data"

## Debug Token

- URL: https://tricorder.pi-hole.net/ytzqrbtz5d

### Screenshots
![2021-06-23_15_35_04](https://user-images.githubusercontent.com/22844936/123107741-19255200-d43a-11eb-984f-5f0f10e4217b.jpg)
![2021-06-23_15_34_13](https://user-images.githubusercontent.com/22844936/123107746-19bde880-d43a-11eb-8b0e-b7637e631be8.jpg)

### Additional context
I run pihole in a container using podman, so in the dashboard my "Top Clients(Total)" is pihole, eventually after a certain amount of queries(between 80,000 and 100,000) are done, when I open the "Top Clients(Total)" pihole on the dashboard to view the queries I get the error "An unknown error occured while loading data" When I then remove the thesqllite db "pihole-FTL.db" and then restart 
the dns using: pihole restartdns, I am then able to view the "Top Client" pihole again. After between 80,0000 and 100,000 queries again  I get the same error message again. Each time this happens currently remove the the pihole-FTL.db and then restart dns but it would be nice if it would just work continuesly.

**Comment by jfb-pihole** [comment]
Please generate a debug log, upload it when prompted and post the token here.

I suspect your device does not have adequate PHP memory to serve the list of queries.  When you clear the database, this removes the query data and allows PHP to load what is now in the database.

**Comment by Hydranet** [comment]
I created a debug token but for some reason, copy/pasting went wrong. I'll have to way till it happens again, which will be tomorrow.  I'll check if I can find where in the container the php.ini settings are located to see at what value it's currently configured.

**Comment by Hydranet** [comment]
I found out why my debug token wasn't showing, I had pasted it in a commented out section of the report, here it is:
[✓] Your debug token is: https://tricorder.pi-hole.net/ytzqrbtz5d

**Comment by DL6ER** [comment]
Your debug log contains many lines like
```
   2021-06-23 12:38:41: (mod_fastcgi.c.421) FastCGI-stderr: PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 8577024 bytes) in /var/www/html/admin/api.php on line 174
   2021-06-23 12:41:00: (mod_fastcgi.c.421) FastCGI-stderr: PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 8609792 bytes) in /var/www/html/admin/api.php on line 174
   2021-06-23 13:57:11: (mod_fastcgi.c.421) FastCGI-stderr: PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 4198400 bytes) in /var/www/html/admin/api.php on line 174
   2021-06-23 14:22:06: (mod_fastcgi.c.421) FastCGI-stderr: PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 1511424 bytes) in /var/www/html/admin/api.php on line 174
   2021-06-23 15:06:58: (mod_fastcgi.c.421) FastCGI-stderr: PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) in /var/www/html/admin/api_FTL.php on line 324
   2021-06-23 15:33:53: (mod_fastcgi.c.421) FastCGI-stderr: PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) in /var/www/html/admin/api_FTL.php on line 324
   2021-06-23 15:38:52: (mod_fastcgi.c.421) FastCGI-stderr: PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) in /var/www/html/admin/api_FTL.php on line 324
   2021-06-23 15:42:14: (mod_fastcgi.c.421) FastCGI-stderr: PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) in /var/www/html/admin/api_FTL.php on line 324
```
so there seems to be another limit configured somewhere on your system.

**Comment by Hydranet** [comment]
Thanks! That is strange, I have no limit configured on my containers and I haven't seen any memory problems on the system where the containers are running. I'm going to see if I can find why I am running into this limit.

**Comment by Hydranet** [comment]
I don't have a memory limit  set on my containers, and went looking in the container, there were only two places where I found a memory limit in php cli and php cgi.
```
7.3/cli/php.ini:memory_limit = -1
7.3/cgi/php.ini:memory_limit = 128M

```
This is the error from the lightttpd log file
```
2021-06-24 16:13:02: (mod_fastcgi.c.421) FastCGI-stderr: PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) in /var/www/html/admin/api_FTL.php on line 324
```
134217728 bytes converts to  134.2MB, which is close enough to 128M so I suspect it's running into the memory limit configured for  cgi php.ini. Is there a way to adjust is or will I have to build my own image using the image already provided?

**Comment by Hydranet** [comment]
Since /etc/php in de container wasn't persistent after I changed it and started lighttpd it was reset to default.  I copied over the /etc/php directory to /etc/pihole and then on my host created a separate directory for it. Which I then mounted with -v in the container and now I'm not getting the error anymore  about "Alllowed memory size of  134217728 bytes exhausted"
So I have it mounted like this in the container:
```
-v "$HOME/pihole/etc-php/php:/etc/php
```
Problem solved, thanks for your time!
